clients/cli/connections.c

```diff
@@ -7764,10 +7772,20 @@ editor_menu_main (NmCli *nmc, NMConnection *connection, const char *connection_t
 					update_connection (persistent, rem_con, update_connection_editor_cb, NULL);
 				}
 
-				//FIXME: add also a timeout for cases the callback is not called
+				nmc_editor_connection_changed = FALSE;
+				handler_id = g_signal_connect (rem_con,
+				                               NM_CONNECTION_CHANGED,
+				                               G_CALLBACK (editor_connection_changed_cb),
+				                               NULL);
```

Comment:
can we do this without globals? Just pass a reference to a variable on the stack as user-data.